### ISSUE TITLE
False positive "Assigning to attribute not defined in class slots" for inherited slots

### ISSUE THREAD

**Issue by noTwanteD** [PyCQA/pylint#731]
The following code generates a false positive E0237:

```python
class Foo(object):
	__slots__ = ['dummy']

	def __init__(self, dummy):
		self.dummy = dummy

class Bar(Foo):
	__slots__ = []

	def Set(self, dummy):
		self.dummy = dummy
```

**Comment by The-Compiler** [comment]
sidenote: please use code blocks like this around code so it doesn't get interpreted as markdown:

    ```python
    [...]
    ```

I edited your post to include them :wink:

**Comment by PCManticore** [comment]
Should be fixed in astroid@master, more exactly in https://github.com/PyCQA/astroid/commit/152087df64cf0ce60dd518f7afbd62e68ac41e80
